src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -68,18 +68,42 @@ public UUID createWorkspace(WorkspaceRequest workspaceRequest, AuthenticatedUser
     return createJob.submitAndWait(UUID.class, true);
   }
 
+  /**
+   * Convenience function that checks existence of a workspace, followed by an authorization check
+   * against that workspace.
+   *
+   * <p>Throws WorkspaceNotFoundException from getWorkspace if the workspace does not exist,
+   * regardless of the user's permission.
+   *
+   * <p>Throws SamUnauthorizedException if the user is not permitted to perform the specified action
+   * on the workspace in question.
+   *
+   * <p>Returns the Workspace object if it exists and the user is permitted to perform the specified
+   * action.
+   *
+   * @param userReq the user's authenticated request
+   * @param workspaceId id of the workspace in question
+   * @param action the action to authorize against the workspace
+   * @return the workspace, if it exists and the user is permitted to perform the specified action.
+   */
+  @Traced
+  public Workspace authorizedGetWorkspace(
+      AuthenticatedUserRequest userReq, UUID workspaceId, String action) {
+    Workspace workspace = workspaceDao.getWorkspace(workspaceId);
+    samService.workspaceAuthzOnly(userReq, workspaceId, action);
```

Comment:
why is the auth check after the `getWorkspace()` call here?

Reply:
this is the core of the entire PR. As discussed in Slack, consensus is that we always want to return 404 if the workspace doesn't exist, regardless of what Sam returns from the permission check. Consider the use case that triggered this PR: deleting a non-existent workspace. In that use case, Sam returns a 403 since it can find no permission for the user. However, we want to return 404. I implemented this logic simply by  putting `getWorkspace()` first, letting it throw 404 if the workspace doesn't exist,  then performing the authz check which  will throw 403  or pass.

Reply:
Please add a comment to that effect.